src/client/Microsoft.Identity.Client/Region/RegionDiscoveryProvider.cs

```diff
@@ -258,7 +258,7 @@ private async Task<Uri> BuildAuthorityWithRegionAsync(Uri canonicalAuthority, Re
             {
                 builder.Host = $"{s_region}.login.microsoft.com";
             }
-            else
+            else if (!builder.Host.Equals($"{s_region}.login.microsoft.com"))
```

Comment:
i would do something like this:
```csharp
 else if (!builder.Host.StartsWith($"{s_region}."))
 {
                builder.Host = $"{s_region}.{builder.Host}";
 }
```

Reply:
@jennyf19's proposal is clearer to me, and this would also work for national clouds: @neha-bhargava said in a previous PR that they were not supported yet, but if they were we'd just prepend the region to the regular hostname, which would work this the suggestion.

Reply:
National clouds are not supported yet, but the spec agreed with estsr mentions this corner case.

Reply:
Anyway, both suggestions seem fine.

Reply:
I agree with what @jmprieur said. For PPE env we just prepend the region to the original authority. Makes sense to implement what @jennyf19 has mentioned.

Reply:
Updated this check to use `StartsWith` and moved it a bit higher, per Neha's suggestion.